File: rust/src/models/runtime_snapshot.rs
use serde::{Deserialize, Serialize};

/// Schema version for the runtime snapshot format.
pub const SNAPSHOT_SCHEMA: &str = "envdoctor.runtime-snapshot.v1";

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct RuntimeSnapshot {
    pub schema: String,
    pub captured_at: String,
    pub os: OsInfo,
    pub tools: Vec<ToolInfo>,
    pub path: Vec<String>,
    pub globals: std::collections::HashMap<String, Vec<GlobalPackage>>,
    pub env_flag_names: Vec<String>,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct OsInfo {
    pub platform: String,
    pub arch: String,
    pub release: String,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ToolInfo {
    pub tool: String,
    pub version: String,
    pub resolved_from: String,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct GlobalPackage {
    pub name: String,
    pub version: String,
}

File: dotnet/src/Envdoctor/Models/RuntimeSnapshot.cs
namespace Envdoctor.Models;

/// Schema version for the runtime snapshot format.
public static class SnapshotSchema
{
    public const string Value = "envdoctor.runtime-snapshot.v1";
}

public sealed class RuntimeSnapshot
{
    public string Schema { get; set; } = SnapshotSchema.Value;
    public string CapturedAt { get; set; } = "";
    public OsInfo Os { get; set; } = new();
    public List<ToolInfo> Tools { get; set; } = new();
    public List<string> Path { get; set; } = new();
    public Dictionary<string, List<GlobalPackage>> Globals { get; set; } = new();
    public List<string> EnvFlagNames { get; set; } = new();
}

public sealed class OsInfo
{
    public string Platform { get; set; } = "";
    public string Arch { get; set; } = "";
    public string Release { get; set; } = "";
}

public sealed class ToolInfo
{
    public string Tool { get; set; } = "";
    public string Version { get; set; } = "";
    public string ResolvedFrom { get; set; } = "";
}

public sealed class GlobalPackage
{
    public string Name { get; set; } = "";
    public string Version { get; set; } = "";
}
